StringUtil.java

```diff
@@ -6,21 +6,66 @@
  * Input:
  */
 public class StringUtil {
-
+    
     private static String[] URLParser(String url){
-        //WRITE YOUR CODE HERE..
-        //parse the url and get all the get parameters and return an array of Strings
-        //Input: google.com/api?param1=value1&param2=value2
-        //Output: array of strings [] = { {param1:value1}, {param2:value2} }
+       String result;
+       String[] tempArray = new String[2];
+       String[] splittedUrl = url.split("[\\?\\&]");
+       String[] resultArray = new String[5];
+       for(int i = 1;i <= splittedUrl.length-1; i++ ){
+           result="";
+           tempArray =splittedUrl[i].split("=");
+           result += "{";
+           result += tempArray[0];
+           result += ":";
+           result += tempArray[1];
+           result += "}";
+           resultArray[i-1] = result;
+       }
+       return resultArray;
 
     }
+    
     private static String reverseWords(String sentence){
-        //WRITE YOUR CODE HERE..
+        Stack wordStack = new Stack();
+        String result = "";
+        if(sentence == null){
+            return sentence;
+        }
+        else {
+            String[] splittedWords = sentence.split(" ");
+            int splitLength = splittedWords.length - 1;
+            for (int i = 0; i <= splitLength; i++) {
+                wordStack.push(splittedWords[i]);
+                if (i < splitLength) {
+                    wordStack.push(" ");
+                }
+            }
+            while (!(wordStack.empty())) {
+                result += wordStack.pop();
+            }
+            return result;
+        }
     }
 
-    private static String reverseString(String sentence){
-        //WRITE YOUR CODE HERE..
+    private static String reverseString(String sentence) {
+        Stack stringStack = new Stack();
+        Queue stringQueue = new LinkedBlockingQueue();
+        String result = "";
+        String[] splittedStrings = sentence.split(" ");
+        int splitLength = splittedStrings.length - 1;
+        for (int i = 0; i <= splitLength; i++) {
+            stringQueue.add((new StringBuffer(splittedStrings[i]).reverse().toString()));
```

Comment:
Avoid creating new objects inside for loops. Too many objects will be created in the heap.
Instead create StringBuffer outside the for loop and assign to it.